docs/.vuepress/components/Search.vue

```diff
@@ -52,67 +58,99 @@
 </template>
 
 <script>
+import debounce from "tiny-debounce";
+import nprogress from "nprogress";
+
 export default {
   props: {
     threshold: {
       type: Number,
       default: 300
     }
   },
-  data () {
+
+  data() {
     return {
       scrollTop: null,
       buckets: [],
       apps: [],
-      query: '',
-      apps_found: false,
-    }
+      query: "",
+      apps_found: false
+    };
   },
-  mounted () {
-    this.axios.get('https://api.github.com/search/repositories?q=topic:scoop-bucket').then((response) => {
-      this.buckets = response.data.items
-    })
+  mounted() {
+    nprogress.configure({ showSpinner: false });
+    nprogress.start();
+    this.axios
+      .get("https://api.github.com/search/repositories?q=topic:scoop-bucket")
+      .then(response => {
+        this.buckets = response.data.items;
+        nprogress.done();
+      });
   },
   methods: {
-    search (value) {
-      let buckets_string = ''
-      this.buckets.forEach((bucket) => {
-        buckets_string += `+repo:${bucket.full_name}`
-      })
-      this.axios.get(`https://api.github.com/search/code?q=${this.query}+in:file+extension:json${buckets_string}`).then((response) => {
-        this.apps = response.data.items
-        this.apps_found = response.data.total_count > 0
-      })
-    },
-    scrollToTop () {
-      window.scrollTo({ top: 0, behavior: 'smooth' })
-      this.scrollTop = 0
+    search: debounce(function() {
+
+      // One char always return zero
+      if (this.query.length < 2) return
+
+      nprogress.start();
+
+      let buckets = "";
+      this.buckets.forEach(bucket => {
```

Comment:
This could be added as a computed method I guess. 
```js
    bucketsList () {
      return '+repo:' + this.buckets.map((el) => {
        return el.full_name
      }).join('+repo:')
    }
```